tests/models/hydrology/test_hydrology_model.py

```diff
@@ -196,181 +197,113 @@ def test_generate_hydrology_model(
 
 
 @pytest.mark.parametrize(
-    "cfg_string, raises",
+    "update_interval, raises",
     [
         pytest.param(
-            "[core]\n"
-            "[core.timing]\nupdate_interval = '1 month'\n"
-            "[hydrology]\ninitial_soil_moisture = 0.5\n"
-            "initial_groundwater_saturation = 0.9\n",
+            pint.Quantity(1, "month"),
             does_not_raise(),
             id="updates correctly",
         ),
         pytest.param(
-            "[core]\n"
-            "[core.timing]\nupdate_interval = '1 week'\n"
-            "[hydrology]\ninitial_soil_moisture = 0.5\n"
-            "initial_groundwater_saturation = 0.9\n",
+            pint.Quantity(1, "week"),
             pytest.raises(NotImplementedError),
             id="incorrect update frequency",
         ),
     ],
 )
 def test_setup(
     dummy_climate_data,
-    cfg_string,
+    fixture_config,
+    update_interval,
     raises,
+    fixture_core_components,
 ):
     """Test set up and update."""
-    from virtual_ecosystem.core.config import Config
     from virtual_ecosystem.core.core_components import CoreComponents
     from virtual_ecosystem.models.hydrology.hydrology_model import HydrologyModel
 
     # Build the config object and core components
-    config = Config(cfg_strings=cfg_string)
-    core_components = CoreComponents(config)
+    fixture_config["core"]["timing"]["update_interval"] = update_interval
+    core_components = CoreComponents(fixture_config)
 
     with raises:
         # initialise model
         model = HydrologyModel.from_config(
             data=dummy_climate_data,
             core_components=core_components,
-            config=config,
+            config=fixture_config,
         )
 
         model.setup()
 
+        # Test soil moisture
+
+        exp_soilm_setup = fixture_core_components.layer_structure.from_template()
+        soil_indices = fixture_core_components.layer_structure.role_indices["all_soil"]
+        exp_soilm_setup[soil_indices] = np.array([[250], [250]])  # [[125], [375]]
```

Comment:
Hmm, something is not quite right here. The values should be different between topsoil and subsoil. I'll see if I can find out where this goes wrong.

Reply:
Ah no, silly me, I was looking at the layer heights instead of the layer thickness. 1-0.5=0.5, so the values should be the same, and should be 250.
You can delete the values in comments

Reply:
Phew - thought I'd radically misunderstood the code 😄